aiida_vasp/calcs/vasp.py

```diff
@@ -342,6 +335,38 @@ def write_wavecar(self, dst, calcinfo):  # pylint: disable=unused-argument
         wave_functions = self.inputs.wavefunctions
         calcinfo.local_copy_list.append((wave_functions.uuid, wave_functions.filename, dst))
 
+    @classmethod
+    def immigrant(cls, code, remote_path, **kwargs):
+        """
+        Returns VaspImmigrant class and associated inputs. This method will be obsolete at v3.0
+
+        :param code: a Code instance for the code originally used.
+        :param remote_path: The directory on the code's computer in which the simulation was run.
+        :param metadata: dict.
+        :param settings: dict. Used for non-default parsing instructions, etc.
+        :param potential_family: str.
+        :param potential_mapping: dict.
+        :param use_wavecar: bool. Try to read WAVECAR.
+        :param use_chgcar bool. Try to read CHGCAR.
+        """
+
+        from aiida_vasp.calcs.immigrant import VaspImmigrant  # pylint: disable=import-outside-toplevel
+
+        proc_cls = VaspImmigrant
+        builder = proc_cls.get_builder_from_folder(code, remote_path, **kwargs)
+        options = {'max_wallclock_seconds': 1, 'resources': {'num_machines': 1, 'num_mpiprocs_per_machine': 1}}
+        metadata = kwargs.get('metadata', {'options': options})
```

Comment:
`label` and `description` as well?

Reply:
I think it's a good point. It is possible to set label and description. But this part is for the backward compatibility to work exactly same as the original manner.
In the new way, setting label and description is encouraged to do out of immigrant, but after getting builder or inputs, the users set them by themselves. This makes the way to use immigrant similar to usual VaspCalculation. This is my design proposal.

Reply:
Sounds good. But maybe it would be okey that new and old way is consistent, while we are anyway working on the code related to `metadata`?

Reply:
I don't see well what you mean the last sentence... Could you tell me a little more here or elsewhere? Thanks.

Reply:
Since we anyway add a `metadata` entry for options, we could add `label` and `description` in case someone should use the old interface, which would be consistent with the new. But if we decide to drop the old in say 3.0 then I think we can leave as is. I am happy to support this. In fact, I think you are the biggest user of immigrant :)

Reply:
I see. Thanks for your explanation. In the next release (such as v2.2), the new immigrant interface will be documented.

Reply:
@atztogo Do you want to not include `label` and `description` at this point? You decide on this.

Reply:
Yes, I pushed the commit of this minor update. Thanks.